docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+---
+title: Keyboard handling
+description: A guide for handling common keyboard interactions.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+
+Keyboard handling is crucial for creating an excellent user experience in your Expo app. This guide covers common keyboard interactions and how to manage them effectively.
+
+React Native provides several keyboard-related APIs, with [`Keyboard`](https://reactnative.dev/docs/keyboard) and [`KeyboardAvoidingView`](https://reactnative.dev/docs/keyboardavoidingview) being the most frequently used. For more complex or custom keyboard interactions, you might consider using [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), which is a library that offers advanced keyboard handling capabilities.
+
+## The basics
+
+### Keyboard events
+
+The `Keyboard` module from React Native allows you to listen for native events, react to them, and make changes to the keyboard, such as dismissing it.
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method accepts an event name and a callback function as arguments. The callback function is called with the event data when the keyboard is shown or hidden.
```

Comment:
Could we ground the keyboard events section in an example of why you'd want to use it? Not that this needs to affect the code example itself, I'm thinking something as simple as "you may want to know if the keyboard is visible or not visible because..."

Searching through some of my old code, one place where I used these events in conjunction with `Keyboard.dismiss` was to set a state variable to show a keyboard dismiss button in the header on a chat screen when they keyboard was visible on iOS, which doesn't have a built-in button for this. Then that button used `Keyboard.dismiss`.

Reply:
or maybe another example would be using a `scrollable view ref to scroll the content to the bottom to show the latest message in a chat?